drivers/clock_control/clock_control_esp32.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 Mohamed ElShahawi.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_CLOCK_CONTROL_ESP32_CLOCK_H_
+#define ZEPHYR_DRIVERS_CLOCK_CONTROL_ESP32_CLOCK_H_
+
+#include <soc/efuse_reg.h>
+
+/* ROM functions which read/write internal control bus */
+extern u8_t esp32_rom_i2c_readReg(u8_t block, u8_t host_id, u8_t reg_add);
+extern void esp32_rom_i2c_writeReg(u8_t block, u8_t host_id, u8_t reg_add, u8_t data);
```

Comment:
Would it make more sense to move these two into `soc.h`? (similar to `esp32_rom_uart_tx_wait_idle()`)

Reply:
I thought about this, but since this is only needed for clock control i put it here.
these functions controls an internal i2c bus for specific functions, may be if there will be another usage for them in another driver, i should move them to soc.h

Reply:
I checked the usage of this internal i2c bus in esp-idf, and i can see it is only used for configuring the PLL and APLL.
So, What Do you think ? should i keep them here or move them to soc.h ?

Reply:
Either way is fine with me. My thought is that since both are also declared via linker script (i.e. `PROVIDE()`), they are sort of globally declared anyway.

Reply:
ok, I moved them now to soc.h with a comment to explain that it is only for internal i2c bus for PLL/APLL